modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -935,10 +935,25 @@
         result              (parent-method driver field-filter)]
     (cond-> result
       field-temporal-type (update :prepared-statement-args (fn [args]
-                                                             (for [arg args]
-                                                               (if (instance? java.time.temporal.Temporal arg)
-                                                                 (->temporal-type field-temporal-type arg)
-                                                                 arg)))))))
+                                                             (map (fn [arg]
+                                                                    (if (instance? java.time.temporal.Temporal arg)
+                                                                      ;; Since we add the zone as part of the
+                                                                      ;; LHS of the filter, we need to add the zone to
+                                                                      ;; the RHS as well.
+                                                                      (let [request-time-zone-id (qp.timezone/requested-timezone-id)
```

Comment:
Probably better to fetch this timezone once outside of the `for`, that way if its value changes inside between evaluation of the lazy sequence returned by `for` it doesn't mess stuff up